bucketlist/views.py

```diff
@@ -0,0 +1,223 @@
+    bucketlist = Bucketlist.query.filter_by(id=bucket_id, created_by=g.user.id).first()
+    if not bucketlist:
+        return jsonify({"Message": "That bucketlist does not exist for your user account. Please try again"}), 404
+    db.session.delete(bucketlist)
+    db.session.commit()
+    return jsonify({"Message": bucketlist.name.title() + " has been deleted"}), 200
+
+
+@app.route("/bucketlists/<int:bucket_id>/items", methods=["POST"])
+@auth_token.login_required
+def new_item(bucket_id):
+    """
+    Creates a new bucketlist item.
+    """
+    # check if bucketlist exists, then validate user input using try-catch block
+    bucketlist = Bucketlist.query.filter_by(id=bucket_id, created_by=g.user.id).first()
+    if not bucketlist:
+        return jsonify({"Message": "You can only create an item within your own bucketlist. Please try again"}), 404
+    else:    
+        try:
+            item = BucketlistItem()
+            sanitized = item.import_data(request.json)
+            if sanitized == "Invalid":
+                return jsonify({"Message": "The item must have a name"}), 400
+        except ValidationError as e:
+            return jsonify({"Message": str(e)}), 400
+
+        # check for duplicates before creating the new item
+        duplicate = BucketlistItem.query.filter_by(name=item.name, bucket=bucket_id).first()
+        if not duplicate:
+            item.bucket = bucket_id
+            item.created_by = g.user.id
+            db.session.add(item)
+            db.session.commit()
+            return jsonify({"Message": item.name.title() + " has been created",
+                            "View it here": item.export_data()}), 201
+        return jsonify({"Message": "A bucketlist item with that name already exists. Please try again"}), 400
+
+
+@app.route("/bucketlists/<int:bucket_id>/items/<int:item_id>", methods=["PUT"])
+@auth_token.login_required
+def update_item(bucket_id, item_id):
+    """
+    Updates a specified item belonging to a specified bucketlist
+    """
+    # ensures that a logged-in user can only access their own bucketlist
+    item = BucketlistItem.query.filter_by(bucket=bucket_id, id=item_id, created_by=g.user.id).first()
+    if not item:
+        return jsonify({"Message": "That item does not exist for your user account. Please try again"}), 404
+    # checks for duplicates before updating the item
+    item.update_data(request.json)
+    db.session.commit()
+    return jsonify({"Message": "Updated: " + item.name.title(),
+                    "View it here": item.export_data()}), 200
+
+
+@app.route("/bucketlists/<int:bucket_id>/items/<int:item_id>", methods=["DELETE"])
+@auth_token.login_required
+def delete_item(bucket_id, item_id):
+    """
+    Deletes a specified item belonging to a specified bucketlist
+    """
+    # ensures that a logged-in user can only access their own bucketlist
+    item = BucketlistItem.query.filter_by(bucket=bucket_id, id=item_id, created_by=g.user.id).first()
```

Comment:
I've tried out your suggested code (it works well without the `.first()`) but it would require me to write additional lines of code to catch the exceptions. The code as currently written has fewer lines of code than if I changed it to incorporate your suggestion, so I'll leave it unchanged, thanks.